### ISSUE TITLE
"error":"invalid_client","error_description":"The audience claim for client_assertion must be the endpoint invoked for the request."

### ISSUE THREAD

**Issue by herkusas** [okta/okta-sdk-dotnet#600]
### Describe the bug?

Hello, I'm trying to migrate from version 5.X.X to version 6.0.0 of OKTA SDK. 

Version 5.X.X code:

         services.AddTransient<IOktaClient>(_ => new OktaClient(
            new OktaClientConfiguration
            {
                OktaDomain = options.BaseUri.AbsoluteUri,
                Scopes = new List<string> {"okta.groups.read", "okta.users.read"},
                ClientId = options.ClientId,
                AuthorizationMode = AuthorizationMode.PrivateKey,
                PrivateKey = new JsonWebKeyConfiguration(JsonSerializer.Serialize(options.PrivateKey))
            }));

         services.AddTransient<IAccessManager>(f => new AccessManager(f.GetRequiredService<IOktaClient>()));

            
Version 6.0.0 code:
 
         services.AddTransient<IGroupApi>(_ => new GroupApi(
            new Configuration
            {
                OktaDomain = options.BaseUri.AbsoluteUri,
                Scopes = new HashSet<string>{"okta.groups.read"},
                ClientId = options.ClientId,
                AuthorizationMode = AuthorizationMode.PrivateKey,
                PrivateKey = new JsonWebKeyConfiguration(JsonSerializer.Serialize(options.PrivateKey))
            }));

        services.AddTransient<IUserApi>(_ => new UserApi(
            new Configuration
            {
                OktaDomain = options.BaseUri.AbsoluteUri,
                Scopes = new HashSet<string>{"okta.users.read"},
                ClientId = options.ClientId,
                AuthorizationMode = AuthorizationMode.PrivateKey,
                PrivateKey = new JsonWebKeyConfiguration(JsonSerializer.Serialize(options.PrivateKey))
            }));

        services.AddTransient<IAccessManager>(f => new AccessManager(f.GetRequiredService<IGroupApi>(), f.GetRequiredService<IUserApi>()));
 
When I try to use any of these two new APIs functions (for example list groups) I get the following error:
 
```
 Okta.Sdk.Client.ApiException: Error calling GetBearerToken: {"error":"invalid_client","error_description":"The audience claim for client_assertion must be the endpoint invoked for the request."}
   at Okta.Sdk.Api.OAuthApi.GetBearerTokenWithHttpInfoAsync(CancellationToken cancellationToken)
   at Okta.Sdk.Api.OAuthApi.GetBearerTokenAsync(CancellationToken cancellationToken)
   at Okta.Sdk.Client.DefaultOAuthTokenProvider.RequestAccessTokenAsync(CancellationToken cancellationToken)
   at Okta.Sdk.Client.DefaultOAuthTokenProvider.GetAccessTokenAsync(Boolean forceRenew, CancellationToken cancellationToken)
   at Okta.Sdk.Client.OktaPagedCollectionEnumerator`1.MoveNextAsync()
  
```
  With prior versions, everything works like a charm. 
  
  If I look into the source code [it looks like that audience value is something I cannot change](https://github.com/okta/okta-sdk-dotnet/blob/master/src/Okta.Sdk/Client/DefaultJwtGenerator.cs#L102)
  
  

### What is expected to happen?

I should be able to get the bearer token and then list groups. 

### What is the actual behavior?

I'm getting an error.

### Reproduction Steps?

Check out the bug description I wrote. 

### .NET Version

6.0.103

### SDK Version

6.0.0

**Comment by laura-rodriguez** [comment]
Hi @herkusas,

Thanks for reporting this issue. I'll file an internal ticket to be reviewed and prioritized by the team.

**Comment by laura-rodriguez** [comment]
Fixed in v6.0.1
